fw/sock.c

```diff
@@ -938,28 +940,32 @@ ss_tcp_data_ready(struct sock *sk)
 	switch (ss_tcp_process_data(sk)) {
 	case SS_OK:
 		return;
-	case SS_BAD:
-		/*
-		 * Close connection in case of internal errors,
-		 * banned packets, or FIN in the received packet,
-		 * and only if it's not on hold until explicitly
-		 * closed.
-		 *
-		 * ss_close() is responsible for calling
-		 * application layer connection closing callback.
-		 * The callback will free all SKBs linked with
-		 * the message that is currently being processed.
-		 *
-		 * Closing a socket should go through the queue and
-		 * should be done after all pending data has been sent.
-		 */
-		if (!(SS_CONN_TYPE(sk) & Conn_Stop))
-			ss_close(sk, SS_F_SYNC);
-		break;
 	case SS_DROP:
-		ss_close(sk, SS_F_ABORT);
+		flags = SS_F_ABORT;
+		break;
+	case SS_BAD:
+		fallthrough;
+	default:
```

Comment:
Which `default` values we can observe here? If this is just a safety, then it make sense to place `WARN_ON_ONCE()` here. The `switch` implements a very subtle logic, so a default path make the things less clear.

Reply:
Fixed